lib/backend/Memory.js

```diff
@@ -272,11 +295,12 @@ export default class Memory {
             const maxScore = (min === '+inf') ?
                 this.data[key][this.data[key].length][0] : max;
             let removeCount = 0;
-            this.data[key].forEach((item, index) => {
+            this.data[key] = this.data[key].filter(item => {
                 if (item[0] >= minScore && item[0] <= maxScore) {
-                    this.data[key].splice(index, 1);
                     removeCount++;
+                    return false;
```

Comment:
Instead of having a side effect in your filter you could check the lengths before and after. But up to you.